### ISSUE TITLE
Syncing media with change file 

### ISSUE THREAD

**Issue by bielu** [Jumoo/uSync.Complete.Issues#254]
**Describe the bug**

**Version (please complete the following information):**
 - Umbraco Version: 13.5.2
 - uSync Version 13.2.5
 - uSync.Complete Version 13.1.8

**To Reproduce**

1. Upload the file on Env1 test1.pdf
2. Use in content and deploy the page and media files to env2 with uSync complete
3. On the same media item in Umbraco on env1, remove the file and upload Test2.pdf
4. Deploy the page and media files to env2 with uSync complete
5. Notice that both urls to files test1.pdf and test2.pdf are still working, so file wasn't removed from blob storage.

**Expected behavior**
After syncing test1.pdf would be removed from blob storage

**Comment by KevinJump** [comment]
Yeah, i will check what this actually looks like in the code .

by default, we don't delete things, they go into the recycle bin, and umbraco doesn't remove things until the bin is empty.

However - I appreciate this is different so i will take a look.

**Comment by bielu** [comment]
@KevinJump the logic of removal of nodes is fine, there is the issue with removal of files attach, so yeah quite different, thanks for looking into it!

**Comment by bielu** [comment]
Hey @KevinJump so any idea if we can fix this?

**Comment by KevinJump** [comment]
so TLDR: - Yes, we can fix this, (and have a working branch, and nightly build if you want to try it).

https://dev.azure.com/jumoo/Public/_artifacts/feed/nightly/NuGet/uSync.Complete/overview/13.1.9-20250108.1

its a bit complicated, because the core bits of uSync which track the changes don't know about media files (they just know about properties and file hash values). and uSync.Complete doesn't know much more than the key and file name of the things its pushing.

So - a bit hacky, we reach into the change details from the sync to get the old and new values of the `umbracoFile` property, we then extract the file source from that (often in a bit of json representing the cropper).

if the old and new source values are different, then we mark the old value for deletion and then after we have imported all the new files we remove these files .

I am, a little bit concerned that if this goes wrong for what ever reason we might remove media files that should be there, but i think it should be ok.

**Comment by KevinJump** [comment]
what we might do is put this behind an option (which might well be on by  default), so it can be turned off if it causes people issues.

**Comment by KevinJump** [comment]
just tiding this up, we are going to have this off by default in v13 (don't want to issue a big breaking change to the 'stable' release. 

"AllowDeletingOfRenamedMediaFiles"

e.g

```json
"uSync": {
  "Publisher": {
    "Settings": {
      "IncomingEnabled": true,
      "AllowDeletingOfRenamedMediaFiles": true,
      "AppId": "[APP_ID]",
      "AppKey": "[APP_KEY]"
    }
  }
}
```

**Comment by bielu** [comment]
hey @KevinJump that's fine, i had opportunity to check it and all seems work as expected, awaiting the official build with new setting! :)

**Comment by KevinJump** [comment]
This is fixed in the latest releases (v13.1.9 / v15.1.1)
